src/transport_picoquic.cpp

```diff
@@ -0,0 +1,828 @@
+  picoquic_set_transport_parameters(cnx, &local_tp_options);
+
+  (void)createStreamContext(cnx, 0);
+
+  return (uint64_t)cnx;
+}
+
+void PicoQuicTransport::client(const TransportContextId tcid)
+{
+  int ret;
+  std::ostringstream log_msg;
+
+  picoquic_cnx_t* cnx = active_streams[tcid][0].cnx;
+
+  log_msg << "Thread client packet loop for client context_id: " << tcid;
+  logger.log(LogLevel::info, log_msg.str());
+
+  if (cnx == NULL) {
+    logger.log(LogLevel::error, "Could not create picoquic connection client context");
+  }
+  else {
+    picoquic_enable_keep_alive(cnx, 3000000);
+    picoquic_set_callback(cnx, pq_event_cb, this);
+
+    ret = picoquic_start_client_cnx(cnx);
+    if (ret < 0) {
+      logger.log(LogLevel::error, "Could not activate connection");
+      return;
+    }
+
+    (void)picoquic_mark_datagram_ready(cnx, true);
+
+    ret = picoquic_packet_loop(quic_ctx, 0, AF_INET, 0,
+                               2000000, 0,
+                               pq_loop_cb, this);
+
+    log_msg.str("");
+    log_msg << "picoquic ended with " << ret;
+    logger.log(LogLevel::info, log_msg.str());
+  }
+
+  if (quic_ctx != NULL) {
+    picoquic_free(quic_ctx);
+    quic_ctx = NULL;
+  }
+
+  shutdown();
+}
+
+void
+PicoQuicTransport::closeStream(const TransportContextId& context_id,
+                               const StreamId stream_id)
+{
+  deleteStreamContext(context_id, stream_id);
+}
+
+void
+PicoQuicTransport::close(
+        [[maybe_unused]] const TransportContextId& context_id)
+{
+}
+
+void PicoQuicTransport::checkDataOut()
+{
+  for (auto& c_pair: active_streams) {
+    for (auto &s_pair : active_streams[c_pair.first]) {
+
+//      std::ostringstream log_msg;
+//      log_msg << "pending data check, ctx: " << c_pair.first
+//              << " stream: " << s_pair.second.stream_id << " size: " << s_pair.second.out_data.size();
+//      logger.log(LogLevel::info, log_msg.str());
+      if (s_pair.second.out_data.size()) {
+          if (s_pair.first == 0) {
+            sendOutData(&s_pair.second, NULL, 1300);
+          } else {
+            (void)picoquic_mark_active_stream(s_pair.second.cnx,
+                                              s_pair.first,
+                                              1,
+                                              (StreamContext*)&s_pair.second);
+          }
+      }
+    }
+  }
+}
+
+void PicoQuicTransport::sendOutData(StreamContext *stream_cnx,
+                                    [[maybe_unused]] uint8_t* bytes_ctx,
+                                    size_t max_len)
+{
+  for (int i=0; i < 120; i++) {
+    const auto& out_data = stream_cnx->out_data.pop();
+    if (out_data.has_value()) {
+//          std::ostringstream log_msg;
+//          log_msg << "Send Data: stream: " << stream_cnx->stream_id
+//                  << " max_len: " << max_len
+//                  << " queue_size: " << stream_cnx->out_data.size();
+//          logger.log(LogLevel::info, log_msg.str());
```

Comment:
Seeing a lot of dead log messages, should we revive them and hide them behind some macro or flag?